Location permission fixes

jobhh opened the pull request:
ACCESS_BACKGROUND_LOCATION was used from API level 21, changed that to API level 29.

ACCESS_COARSE_LOCATION adds no permissions over ACCESS_FINE_LOCATION, so when using the latter the former has no purpose. Removed ACCESS_COARSE_LOCATION.
> If you are using both NETWORK_PROVIDER and GPS_PROVIDER, then you need to request only the ACCESS_FINE_LOCATION permission, because it includes permission for both providers. Permission for ACCESS_COARSE_LOCATION allows access only to NETWORK_PROVIDER.
https://developer.android.com/guide/topics/location/strategies.html#Permission

The permissions were not consistently used (one class only checked coarse, another all 3). I added a util class to keep the behaviour the same.

homeassistant commented:
Hi @jobhh, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant-android#135).

Once you do that we will be able to review and accept this pull request.

Thanks!

JBassett commented:
Seems we have a bit of a back up in PR merges.
https://github.com/home-assistant/home-assistant-android/pull/129 has some of your changes in it.  Just a heads up, not sure who will get merged first.

jobhh commented:
No problem :) 

The other pull request entails more, so it'll probably be easier to merge that one if it is deemed proper and rebase this (if it is still relevant).

robbiet480 commented:
@JBassett I can get stuff merged today, just let me know what should be merged first! Also available on Discord to chat instead.

robbiet480 commented:
#129 has been merged. @jobhh you can rebase on master and get your PR fixed up and then I’ll merge it.

homeassistant commented:
Hello @jobhh,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`bac77097a1b8b5989cabb17093494e4eb7a2147d`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant-android&#x2F;pull&#x2F;135&#x2F;commits&#x2F;bac77097a1b8b5989cabb17093494e4eb7a2147d) - This commit has something that looks like an email address (j.hartjes@dn-mobility.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

jobhh commented:
Rebased. I did first commit with an mail address that isn't connected to github, so I changed the author of that commit later on. Can the automated message that contains the other mail address be removed?

I'm not quite sure why restartLocationTracking is in the PermissionManager, but I decided not to touch it since it's outside of the scope of this pull request :)
